trieste/acquisition/function/function.py

```diff
@@ -699,3 +701,108 @@ def __call__(self, x: TensorType) -> TensorType:
+):
+    """
+    A simple parallelization of the lower confidence bound acquisition function that produces
+    a vectorized acquisition function which can efficiently optimized even for large batches.
+
+    See :cite:`torossian2020bayesian` for details.
+    """
+
+    def __init__(self, search_space: SearchSpace):
+        """
+        :param search_space: The global search space over which the optimisation is defined.
+        """
+        self._search_space = search_space
+
+    def __repr__(self) -> str:
+        """"""
+        return f"MultipleOptimismNegativeLowerConfidenceBound({self._search_space!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+        :return: The multiple optimism negative lower confidence bound function.
+        """
+        return multiple_optimism_lower_confidence_bound(model, self._search_space.dimension)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        tf.debugging.Assert(isinstance(function, multiple_optimism_lower_confidence_bound), [])
+        return function  # nothing to update
+
+
+class multiple_optimism_lower_confidence_bound(AcquisitionFunctionClass):
+    r"""
+    The multiple optimism lower confidence bound (MOLCB) acquisition function for single-objective
+    global optimization.
+
+    Each batch dimension of this acquisiton function correponds to a lower confidence bound
+    acquisition function with different beta values, i.e. each point in a batch chosen by this
+    acquisition function lies on a gradient of exploration/exploitation trade-offs.
+
+    We choose the different beta values following the cdf method of :cite:`torossian2020bayesian`.
+    See their paper for more details.
+    """
+
+    def __init__(self, model: ProbabilisticModel, search_space_dim: int):
+        """
+        :param model: The model of the objective function.
+        :param search_space_dim: The dimensions of the optimisation problem's search space.
+        :raise tf.errors.InvalidArgumentError: If ``search_space_dim`` is not postive.
+        """
+
+        tf.debugging.assert_positive(search_space_dim)
+        self._search_space_dim = search_space_dim
+
+        self._model = model
+        self._initialized = tf.Variable(False)  # Keep track of when we need to resample
+        self._betas = tf.Variable(tf.ones([0], dtype=tf.float64), shape=[None])  # [0] lazy init
+
+    @tf.function
+    def __call__(self, x: TensorType) -> TensorType:
+
+        batch_size = tf.shape(x)[-2]
```

Comment:
is -2 relying on x being rank=2? if so can we make it more flexible?

Reply:
just assuming the the batch dim is the penultimate one. This is assumed throughout our code base. Arbitrary leading dims would be fine :)